src/Components/CartModal/CartModal.js

```diff
@@ -0,0 +1,59 @@
+import React, { useEffect, useState, useRef } from "react";
+import "./CartModal.scss";
+
+function CartModal() {
+  const outside = useRef();
+  const [isOpen, setIsOpen] = useState(false);
+  const handleClick = (e) => {
+    if (outside.current.contains(e.target)) {
+      return;
+    }
+    setIsOpen(false);
+  };
+
+  useEffect(() => {
+    const getClick = document.addEventListener("click", handleClick);
+
+    return () => {
+      getClick();
+    };
+  }, []); //empty bracket makes it run onetime,
+
+  return (
+    <div className="CartModal" ref={outside}>
+      <button onClick={() => setIsOpen(!isOpen)}>장바구니 모달창 !!!!</button>
+
+      {isOpen ? (
```

Comment:
```suggestion
      {isOpen && (
      ...
      }
```
true 조건만 사용할 때는 && 사용해주세요